multipaz-models/src/commonTest/kotlin/org/multipaz/models/presentment/digitalCredentialsPresentmentTest.kt

```diff
@@ -0,0 +1,256 @@
+            request: MdocRequest,
+            document: Document,
+        ): Boolean {
+            for (credential in document.getCertifiedCredentials()) {
+                if (credential is MdocCredential && credential.docType == request.docType) {
+                    return true
+                }
+            }
+            return false
+        }
+
+        private suspend fun sdjwtFindDocumentsForRequest(
+            request: VcRequest,
+        ): List<Document> {
+            val now = Clock.System.now()
+            val result = mutableListOf<Document>()
+
+            for (documentName in documentStore.listDocuments()) {
+                val document = documentStore.lookupDocument(documentName) ?: continue
+                if (sdjwtDocumentMatchesRequest(request, document)) {
+                    result.add(document)
+                }
+            }
+            return result
+        }
+
+        private suspend fun sdjwtDocumentMatchesRequest(
+            request: VcRequest,
+            document: Document,
+        ): Boolean {
+            for (credential in document.getCertifiedCredentials()) {
+                if (credential is SdJwtVcCredential && credential.vct == request.vct) {
+                    return true
+                }
+            }
+            return false
+        }
+
+        override suspend fun getCredentialForPresentment(
+            request: Request,
+            document: Document
+        ): CredentialForPresentment {
+            return when (request) {
+                is MdocRequest -> mdocGetCredentialsForPresentment(request, document)
+                is VcRequest -> sdjwtGetCredentialsForPresentment(request, document)
+            }
+        }
+
+        private suspend fun mdocGetCredentialsForPresentment(
+            request: MdocRequest,
+            document: Document,
+        ): CredentialForPresentment {
+            val now = Clock.System.now()
+            return CredentialForPresentment(
+                credential = document.findCredential(domain = "mdoc", now = now),
+                credentialKeyAgreement = null
+            )
+        }
+
+        private suspend fun sdjwtGetCredentialsForPresentment(
+            request: VcRequest,
+            document: Document,
+        ): CredentialForPresentment {
+            val now = Clock.System.now()
+            return CredentialForPresentment(
+                credential = document.findCredential(domain = "sdjwt", now = now),
+                credentialKeyAgreement = null
+            )
+        }
+
+        override fun shouldShowConsentPrompt(
+            credential: Credential,
+            request: Request
+        ): Boolean = true
+
+        override fun shouldPreferSignatureToKeyAgreement(
+            document: Document,
+            request: Request
+        ): Boolean = true
+    }
+
+
+    @Test
+    fun openid4vpHappyPath() = runTest {
+        documentStoreTestHarness.initialize()
+
+        /*
```

Comment:
Remove commented-out code or add a comment here indicating why it's commented out and when we might uncomment it.

Reply:
Done